caikit/runtime/http_server/http_server.py

```diff
@@ -318,19 +341,34 @@ def stop(self):
         if self.interrupter:
             self.interrupter.stop()
 
-    ##########
-    ## Impl ##
```

Comment:
I did a lot of rearranging here. Sorry! The `Impl` section had too many functions of different flavors, so I broke it up into `Static Endpoints`, `Request Binding`, and `Helpers`.

Reply:
🌶️🌶️🌶️

I like the organization! While we're at it, can we take this to the logical conclusion of splitting out http servicers for separate tasks? This file is getting suuuuper chonky with everything still piled into one class.

Since the new endpoint is static, I actually expected to see something more like this:

```
class HttpModelManagermentServicer:

    @app.post(MODEL_MANAGEMENT_ENDPOINT, ...)
    async def deploy_model(self, stuff...)
        # do the deploy things
```

Related question: Why are this and the training management endpoints not in the `Static Endpoints` section here as is? I thought they were static, as opposed to the dynamically registered inference and training endpoints that depend on the caikit library implementation.

Reply:
>Related question: Why are this and the training management endpoints not in the Static Endpoints section here as is? I thought they were static, as opposed to the dynamically registered inference and training endpoints that depend on the caikit library implementation.

Very valid question. The reason I did them as dynamic methods was the need to compute the input/output types from data objects. Technically, I could have just hard-coded the data objects because I _do_ know them, but this would create multiple places where the definition of the service is explicitly coded (service definition and here). I preferred to keep the service definition as the single source-of-truth so that any changes down the line do not require mucking with these handlers.

>While we're at it, can we take this to the logical conclusion of splitting out http servicers for separate tasks? This file is getting suuuuper chonky with everything still piled into one class.

Yeah, we probably _should_ do this, but I'm afraid it would make this PR even more of a "general purpose refactor" which was not its intent. I'd prefer to open a ticket and revisit in a separate PR.